src/core/public/application/types.ts

```diff
@@ -252,7 +255,7 @@ export interface ApplicationStart {
    * @param contextName - The key of {@link AppMountContext} this provider's return value should be attached to.
    * @param provider - A {@link IContextProvider} function
    */
-  registerMountContext<T extends keyof AppMountContext>(
+  registerMountContext<T extends AppMountContextNames>(
```

Comment:
We shouldn't utilize once calculated types for extendable types.
![2019-08-13_12-18-01](https://user-images.githubusercontent.com/3198181/62934383-b4702980-bdc4-11e9-8f4d-d50e4166034d.png)